pkg/cmd/container/run_cgroup_linux.go

```diff
@@ -40,73 +39,72 @@ type customMemoryOptions struct {
 	disableOOMKiller  *bool
 }
 
-func generateCgroupOpts(cmd *cobra.Command, globalOptions types.GlobalCommandOptions, id string) ([]oci.SpecOpts, error) {
-	cpus, err := cmd.Flags().GetFloat64("cpus")
-	if err != nil {
-		return nil, err
-	}
-	memStr, err := cmd.Flags().GetString("memory")
-	if err != nil {
-		return nil, err
-	}
-	memSwap, err := cmd.Flags().GetString("memory-swap")
-	if err != nil {
-		return nil, err
+func generateCgroupPath(id string, cgroupManager, cgroupParent string) (string, error) {
```

Comment:
nit: can we keep the argument order unchanged and remove the redundant `string` arg type (`cgroupManager, cgroupParent, id string). Also if you move this func to after `GenerateOpts` and before `ParseDevice`, the diff will be much smaller.